backends/imgui_impl_dx9.cpp

```diff
@@ -70,6 +75,435 @@ static ImGui_ImplDX9_Data* ImGui_ImplDX9_GetBackendData()
+            };
+            
+            PS_Output main(PS_Input input)
+            {
+                PS_Output output;
+                
+                output.col = input.col * tex2D(tex0, input.uv);
+                
+                return output;
+            };
+        */
+        // Precompile with Visual Studio 2019 HLSL compiler, ps_2_0
+        CODE(00,02,FF,FF),
+        CODE(FE,FF,1F,00),
+        CODE(43,54,41,42),
+        CODE(1C,00,00,00),
+        CODE(4F,00,00,00),
+        CODE(00,02,FF,FF),
+        CODE(01,00,00,00),
+        CODE(1C,00,00,00),
+        CODE(00,01,00,00),
+        CODE(48,00,00,00),
+        CODE(30,00,00,00),
+        CODE(03,00,00,00),
+        CODE(01,00,02,00),
+        CODE(38,00,00,00),
+        CODE(00,00,00,00),
+        CODE(74,65,78,30),
+        CODE(00,AB,AB,AB),
+        CODE(04,00,0C,00),
+        CODE(01,00,01,00),
+        CODE(01,00,00,00),
+        CODE(00,00,00,00),
+        CODE(70,73,5F,32),
+        CODE(5F,30,00,4D),
+        CODE(69,63,72,6F),
+        CODE(73,6F,66,74),
+        CODE(20,28,52,29),
+        CODE(20,48,4C,53),
+        CODE(4C,20,53,68),
+        CODE(61,64,65,72),
+        CODE(20,43,6F,6D),
+        CODE(70,69,6C,65),
+        CODE(72,20,31,30),
+        CODE(2E,31,00,AB),
+        CODE(1F,00,00,02),
+        CODE(00,00,00,80),
+        CODE(00,00,03,B0),
+        CODE(1F,00,00,02),
+        CODE(00,00,00,80),
+        CODE(00,00,0F,90),
+        CODE(1F,00,00,02),
+        CODE(00,00,00,90),
+        CODE(00,08,0F,A0),
+        CODE(42,00,00,03),
+        CODE(00,00,0F,80),
+        CODE(00,00,E4,B0),
+        CODE(00,08,E4,A0),
+        CODE(05,00,00,03),
+        CODE(00,00,0F,80),
+        CODE(00,00,E4,80),
+        CODE(00,00,E4,90),
+        CODE(01,00,00,02),
+        CODE(00,08,0F,80),
+        CODE(00,00,E4,80),
+        CODE(FF,FF,00,00),
+    };
+    #undef CODE
+    if (D3D_OK != bd->pd3dDevice->CreateVertexShader((DWORD*)VertexShaderByteCode, &bd->pVertexShader))
+        return false;
+    if (D3D_OK != bd->pd3dDevice->CreatePixelShader((DWORD*)PixelShaderByteCode, &bd->pPixelShader))
+        return false;
+    
+    bd->IsShaderPipeline = true;
+    ImGui::GetIO().BackendRendererName = "imgui_impl_dx9 (shader)";
+    return true;
+}
+static void ImGui_ImplDX9_Shader_DestroyPipeline()
+{
+    ImGui_ImplDX9_Data* bd = ImGui_ImplDX9_GetBackendData();
+    if (bd->pInputLayout) { bd->pInputLayout->Release(); bd->pInputLayout = NULL; }
+    if (bd->pVertexShader) { bd->pVertexShader->Release(); bd->pVertexShader = NULL; }
+    if (bd->pPixelShader) { bd->pPixelShader->Release(); bd->pPixelShader = NULL; }
+    bd->IsShaderPipeline = false;
+}
+static bool ImGui_ImplDX9_Shader_SetRenderState(ImDrawData* draw_data)
+{
+    ImGui_ImplDX9_Data* bd = ImGui_ImplDX9_GetBackendData();
+    if (!bd->IsShaderPipeline)
+        return false;
+    
+    IDirect3DDevice9* ctx = bd->pd3dDevice;
+    
+    // Setup orthographic projection matrix
```

Comment:
Could this function call regular `ImGui_ImplDX9_SetRenderState()` and then add the extra stuff afterwise?